src/components/CarInfo.component.tsx

```diff
@@ -0,0 +1,160 @@
+import { useMutation } from "@apollo/client";
+import React, { useContext, useEffect, useState } from "react";
+import { useNavigate } from "react-router-dom";
+import { toast } from "react-toastify";
+import "react-toastify/dist/ReactToastify.css";
+import styled from "styled-components";
+
+import { UserContext } from "../contexts/UserContext";
+import { User_Cars } from "../graphql/generated/graphql";
+import { CREATE_USER_CAR, DELETE_USER_CAR } from "../graphql/mutations";
+import { deleteCarVariables } from "../graphql/variables";
+import { CarRowInfo } from "../interfaces/Car";
+import { defaultTheme } from "../theme";
+import { PATHNAME } from "../utils";
+import { FavoriteButton } from "./FavoriteButton.component";
+import { FlexColumn, FlexRow, H4, P } from "./styled";
+
+export const CarInfo = ({ img, car }: { img: string; car: CarRowInfo }) => {
+  const { loggedUser } = useContext(UserContext);
+  const [isFav, setIsFavorite] = useState<boolean>(car.isFavorite ?? false);
```

Comment:
why do you need the `??` operator?

Reply:
since that state (isFav) is a boolean I cannot initialize the state with a null or undefined, so `??` operator is helping me cover the cases where `car.isFavorite` is null or undefined and in that cases initialize the state as false.